File: Selenium/Activities/Activity7.py
# Import webdriver from selenium
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.firefox.service import Service as FirefoxService
from webdriver_manager.firefox import GeckoDriverManager
from selenium.webdriver.common.action_chains import ActionChains

# Set up the Firefox Driver with WebDriverManger
service = FirefoxService(GeckoDriverManager().install())

# Start the Driver
with webdriver.Firefox(service=service) as driver:
    # Navigate to the URL
    driver.get("https://www.training-support.net/selenium/input-events")
    # Print the title of the page
    print("Page title is: ", driver.title)
    # Set up the actions object
    actions = webdriver.ActionChains(driver)

    # Find the football
    ball = driver.find_element(By.ID, "draggable")
    # Find the dropzones1 and dropzone2
    dropzone1 = driver.find_element(By.ID, "droppable")
    dropzone2 = driver.find_element(By.ID, "dropzone2")

    # Perform drag and drop to dropzone 1
    actions.drag_and_drop(ball, dropzone1).perform()
    if dropzone1.text.find("Dropped!"):
        print("Dropzone 1 was activated!")

    # Perform drag and drop to dropzone 2
    actions.drag_and_drop(ball, dropzone2).perform()
    if dropzone2.text.find("Dropped!"):
        print("Dropzone 2 was activated!")
    # Close the browser
    driver.quit()

File: Selenium/Activities/Activity7.java
import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.interactions.Actions;

public class Activity7 {
    public static void main(String[] args) {
        // Set up Firefox driver
        WebDriverManager.firefoxdriver().setup();
        // Create a new instance of the Firefox driver
        WebDriver driver = new FirefoxDriver();
        // Create the Actions object
        Actions builder = new Actions(driver);
        // Open the page
        driver.get("https://training-support.net/selenium/drag-drop");
        // Print the title of the page
        System.out.println("Home page title: " + driver.getTitle());
        // Find the football
        WebElement football = driver.findElement(By.id("draggable"));
        // Find the dropzone1
        WebElement dropzone1 = driver.findElement(By.id("droppable"));
        // Find the dropzone2
        WebElement dropzone2 = driver.findElement(By.id("dropzone2"));
        // Perform drag and drop to dropzone 1
        builder.clickAndHold(football).moveToElement(dropzone1).pause(2000).release().build().perform();
        if(dropzone1.getText()=="Dropped!")
            System.out.println("Dropzone 1 was activated!");
        // Perform drag and drop to dropzone 2
        builder.dragAndDrop(football, dropzone2).build().perform();
        if(dropzone2.getText()=="Dropped!")
            System.out.println("Dropzone 2 was activated!");
        // Close the browser
        driver.close();
    }
}



